### PR TITLE
feat: BI-5952 file-uploader-api handler for processing presigned file from bucket

### PR DESCRIPTION
**Pull request by KonstantAnxiety**
Had to switch to minio for s3 because cloudserver does not implement POST upload with presigned url

Also foreshadowing issues with presigned URLs, as far as I've tested, presigned URL region does not matter, but this could be just for YC where there is only one region, so I think we should add region name to our S3 settings in future

### CODE REVIEW

File: lib/dl_s3/dl_s3/utils.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from typing import TYPE_CHECKING
+
+import botocore.exceptions
+
+if TYPE_CHECKING:
+    from types_aiobotocore_s3 import S3Client as AsyncS3Client
+
+
+async def s3_file_exists(s3_client: AsyncS3Client, bucket: str, key: str) -> bool:
```
Comment (KonstantAnxiety) [inline lib/dl_s3/dl_s3/utils.py:11]: TODO move the rest of s3_utils here

**Comment by KonstantAnxiety** [comment]
TODO

Going to move the rest of [s3_utils](https://github.com/datalens-tech/datalens-backend/pull/762/files#diff-ff2f7039c44d4de6afbe5ad54637c2095b67e9abdf907870708195b12c99ad21) to [dl_s3](https://github.com/datalens-tech/datalens-backend/pull/762/files#diff-63a4e312cee76747fc474580441ef8ea9be8df6d901679779d9267ac117e005a) and switch to minio in all packages in a separate PR

File: lib/dl_file_uploader_api_lib/docker-compose.yml [hunk 2]
```diff
@@ -11,15 +11,14 @@ services:
...
     environment:
-      S3BACKEND: "mem"
-      REMOTE_MANAGEMENT_DISABLE: 1
+      MINIO_ROOT_USER: accessKey1
+      MINIO_ROOT_PASSWORD: verySecretKey1
+      MINIO_DOMAIN: local
+    command: server /export
     ports:
-      - 51420:8000
+      - "51420:9000"
```
Comment (ovsds) [inline lib/dl_file_uploader_api_lib/docker-compose.yml:21]: why port change?

File: lib/dl_file_uploader_api_lib/docker-compose.yml [hunk 3]
```diff
@@ -11,15 +11,14 @@ services:
...
-      dockerfile: Dockerfile.s3-storage
-    command: bash /data/entrypoint.sh
+    image: minio/minio:RELEASE.2024-12-18T13-15-44Z@sha256:1dce27c494a16bae114774f1cec295493f3613142713130c2d22dd5696be6ad3
     environment:
-      S3BACKEND: "mem"
-      REMOTE_MANAGEMENT_DISABLE: 1
+      MINIO_ROOT_USER: accessKey1
+      MINIO_ROOT_PASSWORD: verySecretKey1
+      MINIO_DOMAIN: local
+    command: server /export
```
Comment (ovsds) [inline lib/dl_file_uploader_api_lib/docker-compose.yml:19]:
```suggestion
    command: server --address ":8000" /export
```
suggestion: you can change minio port, so you don't need to change port anywhere else?

**Review by ovsds — DISMISSED** [review]

**Review by ovsds — APPROVED** [review]

**Pull request merged**